### ISSUE TITLE
An question about AFL algorithm

### ISSUE THREAD

**Issue by Zhg9300** [litian96/fair_flearn#15]
Hello, thanks for sharing your wonderful codes.
Recently I’m trying to reproduce the results in the paper FAIRRESOURCEALLOCATION INFEDERATED LEARNING. While plotting the right figure on Figure 9(q-FFL is more efficient than AFL), I failed to get a similar plot as given. No matter how I tune the parameters, the lowest accuracy among all devices remains nearly 20% for AFL algorithm even after 80 rounds of training. Could you please tell me how to set the two parameters $\gamma_w$ and $\gamma_\lambda$ ?

**Comment by trishaladabala** [comment]
Hi there! Thanks for your interest in reproducing these results.

To fix the issue you're experiencing with the lowest accuracy remaining near 20%, you'll need to use the specific hyperparameters mentioned in the original paper's appendix for the Fashion MNIST dataset.

The correct parameters to use for AFL are:

* γw (`--learning_rate`): **0.001**
* γλ (`--learning_rate_lambda`): **0.01**

Here is the exact command you can run:

```bash
python main.py \
    --dataset=fmnist \
    --optimizer=afl \
    --model=lr \
    --learning_rate=0.001 \
    --learning_rate_lambda=0.01 \
    --num_rounds=80 \
    --eval_every=1 \
    --clients_per_round=3 \
    --batch_size=10 \
    --q=0 \
    --sampling=2 \
    --num_epochs=1 \
    --data_partition_seed=0 \
    --log_interval=10 \
    --track_individual_accuracy=1 \
    --static_step_size=0 \
    --output="./log_fmnist/afl_fmnist_results"
```

With these parameters, by round 80 the worst-performing device (Client 6: shirt) should already exceed **40% accuracy**, rather than remaining near the 20% plateau you observed.

Also, according to the repository's `README.md`, reproducing the full convergence shown in Figure 9 for Fashion MNIST requires running training for **6,000 rounds** rather than only 80 rounds. Running the experiment for the full duration with the above hyperparameters should allow the worst-performing device to eventually reach the approximately **71% accuracy** reported in the paper.

Hope this helps!
